antarest/core/interfaces/service.py

```diff
@@ -3,16 +3,28 @@
 
 
 class IService(ABC):
-    def __init__(self) -> None:
-        self.thread = threading.Thread(
-            target=self._loop,
-            name=self.__class__.__name__,
-            daemon=True,
-        )
+    """
+    A base class for long running processing services.
+
+    Processing may be started either in a background thread or in current thread.
+    Implementations must implement the `_loop` method.
+    """
 
     def start(self, threaded: bool = True) -> None:
+        """
+        Starts the processing loop.
+
+        Args:
+            threaded: if True, the loop is started in a daemon thread,
+                      else in this thread
+        """
         if threaded:
-            self.thread.start()
+            thread = threading.Thread(
```

Comment:
@laurent-laporte-pro in agreement with one of your fixme, I removed the reference to thread in "self".

Note that the threading module keeps references to all active threads, so there should be no issue to not hold any reference to it on our side.